weave/Scripts/Main.cs

```diff
@@ -61,6 +95,25 @@ private void SpawnPlayers()
         });
     }
 
+    private bool IsIntersecting(Player player, List<SegmentShape2D> segments)
+    {
+        var position = player.GlobalPosition;
+        var circleShape = (CircleShape2D)player.collisionShape2D.Shape;
+        var radius = circleShape.Radius + lineWidth / 2;
+
+        for (var i = 0; i < segments.Count; i++)
```

Comment:
We can convert this to a fun LINQ expression `return segments.Any(t => Geometry2D.SegmentIntersectsCircle(t.A, t.B, position, radius) != -1);`